crypto/asn1/a_gentm.c

```diff
@@ -22,7 +22,7 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
     static const int min[9] = { 0, 0, 1, 1, 0, 0, 0, 0, 0 };
     static const int max[9] = { 99, 99, 12, 31, 23, 59, 59, 12, 59 };
```

Comment:
Hmm, I dug a little on 'leap second' and I don't think '23:59:60' is valid with all date. In fact, there are only specific dates in which 'leap second' is valid, such as 2016.12.31, 2015.6.30... So I suspect that we may need extra efforts to check if 'date' is also valid instead of just increasing the max-second, otherwise we may generate a certificate with the validity date '2017.6.1 23:59:60' which is definitely an invalid date/time.